ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/AddressIndexedSignerProvider.java

```diff
@@ -62,7 +62,11 @@ public static AddressIndexedSignerProvider create(final SignerProvider signerPro
   public Optional<Signer> getSigner(final String address) {
     final ECPublicKey publicKey = addressToPublicKeyMap.get(address.toLowerCase());
     if (publicKey == null) {
-      return Optional.empty();
+      // attempts to load config/signer via address
+      final Optional<Signer> signer = signerProvider.getSigner(address.toLowerCase());
+      signer.ifPresent(
```

Comment:
Could we just call availablePublicKeys, update our map and if it's in the addressToPublicKeyMap try and grab the new signerProvider? Not sure how expensive the call for availablePublicKeys is though. But it's also exposed directly so might not be an issue.